http-server/src/test/java/io/airlift/http/server/DummyServlet.java

```diff
@@ -15,33 +15,64 @@
  */
 package io.airlift.http.server;
 
-import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.util.Objects.requireNonNull;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
 
 class DummyServlet
         extends HttpServlet
 {
+    private final Optional<CountDownLatch> latch;
+    private AtomicLong requestsReceived = new AtomicLong();
+    public DummyServlet(Optional<CountDownLatch> latch)
```

Comment:
Since this is only used when you have a latch, I'd change this to take a `CountDownLatch` directly.